src/NodaTime.Demo/OffsetDateTimeDemo.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright 2019 The Noda Time Authors. All rights reserved.
+// Use of this source code is governed by the Apache License 2.0,
+// as found in the LICENSE.txt file.
+
+using NUnit.Framework;
+
+namespace NodaTime.Demo
+{    
+    public class OffsetDateTimeDemo
+    {
+        [Test]
+        public void Construction()
+        {
+            LocalDateTime localDateTime = new LocalDateTime(1985, 10, 26, 1, 18);
+            Offset offset = Offset.FromHours(-5);
+            OffsetDateTime offsetDateTime = Snippet.For(new OffsetDateTime(localDateTime, offset));
+
+            Assert.AreEqual(localDateTime, offsetDateTime.LocalDateTime);
+            Assert.AreEqual(offset, offsetDateTime.Offset);
+        }
+
+        [Test]
+        public void WithCalendar()
+        {
+            LocalDateTime localDateTime = new LocalDateTime(1985, 10, 26, 1, 18, CalendarSystem.Gregorian);
+            Offset offset = Offset.FromHours(-5);
+            OffsetDateTime original = new OffsetDateTime(localDateTime, offset);
+            OffsetDateTime updated = Snippet.For(original.WithCalendar(CalendarSystem.Iso));
+
+            Assert.AreNotEqual(original.LocalDateTime, updated.LocalDateTime);
+            Assert.AreEqual(original.Offset, updated.Offset);
+            Assert.AreEqual(CalendarSystem.Iso, updated.Calendar);
+        }
+
+        [Test]
+        public void WithOffset()
+        {
+            LocalDateTime localDateTime = new LocalDateTime(1985, 10, 26, 1, 18);
+            Offset offset = Offset.FromHours(-5);
+            OffsetDateTime original = new OffsetDateTime(localDateTime, offset);
+            OffsetDateTime updated = Snippet.For(original.WithOffset(Offset.FromHours(-3)));
+
+            Assert.AreNotEqual(original.LocalDateTime, updated.LocalDateTime);
```

Comment:
Again, construct the expected date/time and use Assert.AreEqual. You probably don't need to call ToString explicitly this time though, as the difference will be obvious enough.